ddtrace/opentelemetry/span.go

```diff
@@ -70,39 +71,43 @@ func (s *span) extractTraceData(c *oteltrace.SpanContextConfig) {
 	}
 	state, err := oteltrace.ParseTraceState(headers["tracestate"])
 	if err != nil {
-		c.TraceState = state
+		log.Debug("Couldn't parse tracestate: %v", err)
 	}
+	c.TraceState = state
 	parent := strings.Trim(headers["traceparent"], " \t-")
-	if len(parent) != 0 {
+	if len(parent) > 3 {
+		// checking the length to avoid panic when parsing
 		if f, err := strconv.ParseUint(parent[len(parent)-3:], 16, 8); err != nil {
 			c.TraceFlags = oteltrace.TraceFlags(f)
+		} else {
+			log.Debug("Couldn't parse traceparent: %v", err)
 		}
 	}
+	// Remote indicates a remotely-created Span
 	c.Remote = true
 }
 
+// todo: verify whether we need
 func uint64ToByte(n uint64, b []byte) {
-	binary.LittleEndian.PutUint64(b, n)
+	binary.BigEndian.PutUint64(b, n)
```

Comment:
Ideally we should have a test that would fail if we incorrectly switch this back to LittleEndian in the future. Can you add one?

Reply:
Sure, will add

Reply:
In a separate PR alongside other tests